### DISCUSSION TITLE
Serializing into a non-string type when using createAsyncStoragePersister()?

### DISCUSSION THREAD

**Discussion by ccreight** [TanStack/query#8160] [Ideas]
With the current implementation of createAsyncStoragePersister(), the objects being stored must serialize into and be deserialized out of a string type, which causes issues with Sets and Maps, which become empty objects. Is handling non-string types something that could be supported, or if that's needed for a project is it expected that you'd just create a new custom persister?

**Comment by TkDodo** [comment]
You can pass your custom serializer / deserializer function to handle those cases.
